pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -298,8 +298,9 @@ func convertInstances(cfg model.Config, services []*model.Service) []*model.Serv
 	}
 	for _, service := range services {
 		for _, serviceEntryPort := range serviceEntry.Ports {
-			if len(serviceEntry.Endpoints) == 0 &&
+			if len(serviceEntry.Endpoints) == 0 && serviceEntry.WorkloadSelector == nil &&
 				serviceEntry.Resolution == networking.ServiceEntry_DNS {
+				// Note: only convert the hostname to service instance if WorkloadSelector unset
```

Comment:
/s/if WorkloadSelector unset/if WorkloadSelector is not set

Reply:
ok